src/main.cpp

```diff
@@ -158,13 +205,13 @@ static bool _get_options(int argc, char **argv, Options &options)
 		 "using one.")
 
 		("hsm-pin,P", po::value<string>(&options.hsm_pin),
-		 "The PKCS#11 PIN to set up on the HSM, if using one.");
+		 "The PKCS#11 PIN to set up on the HSM, if using one.")
+
+		("factory,f", po::value<string>(&options.factory),
```

Comment:
I have mixed feeling about exposing `factory` command line param. Customers may start abusing it and cross-factory registrations.

Reply:
it was already possible, so it was just security theater. They can't abuse unless they belong to the factory. 

However, I would read /etc/os-release up front so that you can pre-populate the default factory value in this usage message. That way a user doesn't the impression they need to specify it.

Reply:
It was no such CLI param before.

Reply:
correct, there wasn't. I can hide it if it's an issue

Reply:
yes, but in the past it was impossible to not have a default factory defined. 

However - a more pragmatic view is this: We have a couple of partners who are now wanting to release devices with a common base image and then let a customer do an `lmp-device-regster` into the the *customers* factory. This option would make that UX a little cleaner.